### PR TITLE
Removal of bazel/$package:$tag name format requirement

### PR DESCRIPTION
**Pull request by ArthurRab**
Removing assumption of image names being in the format bazel/$package:$tag by changing the shell scripts to extract the image ID from the .tar and use that instead. (Applied to download_pkgs, install_pkgs, and add_apt_key rules)

### CODE REVIEW

File: util/run.bzl [hunk 1]
```diff
@@ -53,7 +53,7 @@ def _extract_impl(ctx, name="", image_tar=None, image_name="", commands=None, ex
         output=script,
         substitutions={
           "%{load_statement}": load_statement,
-          "%{image}": image_name,
```
Comment (xingao267) [inline util/run.bzl]: We can probably remove image_name attr from _run_and_extract rule as it's private. The public macro doesn't ask for image_name. So it should be safe. Please double check.

File: package_managers/download_pkgs.bzl [hunk 2]
```diff
@@ -72,24 +72,36 @@ def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_exec
...
-    builder_image_name = "bazel/%s:%s" % (image_tar.owner.package,
-                                          image_tar.basename.split(".tar")[0])
-
     # Generate a shell script to run apt_get inside this docker image.
     # TODO(tejaldesai): Replace this by docker_run rule
     build_contents = """\
 #!/bin/bash
 set -ex
-docker load --input {image_tar}
+#Find the image's ID
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply package_managers/download_pkgs.bzl]: Done 1 and 2

File: package_managers/download_pkgs.bzl [hunk 3]
```diff
@@ -72,24 +72,36 @@ def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_exec
...
+mkdir temp_for_extracting_id
+cd temp_for_extracting_id
+tar -x < ../{image_tar}
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply package_managers/download_pkgs.bzl]: Done

File: package_managers/download_pkgs.bzl [hunk 4]
```diff
@@ -72,24 +72,36 @@ def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_exec
...
+i=1
+while [ true ]
+do
+    if [ $(cut -d '"' -f$i manifest.json) = "Config" ]
+    then   
+        image_ID=$( cut -d '"' -f$(expr $i + 2) manifest.json | cut -d "." -f1)
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply package_managers/download_pkgs.bzl]: Done (all files)

File: package_managers/download_pkgs.bzl [hunk 5]
```diff
@@ -72,24 +72,36 @@ def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_exec
...
+        break
+    fi
+    i=`expr $i + 1`
+
+done
+cd ..
+rm -rf temp_for_extracting_id
+docker load < {image_tar}
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply package_managers/download_pkgs.bzl]: Accident. Reverted.

File: util/extract.sh.tpl [hunk 6]
```diff
@@ -2,9 +2,30 @@
 
 set -ex
 
+mkdir temp_for_extracting_id
+cd temp_for_extracting_id
+tar -x < ../%{image_tar}
+i=1
+while [ true ]
+do
+
```
[the remark this answers is not in the document]
Comment (ArthurRab) [reply util/extract.sh.tpl]: Fixed

File: util/run.bzl (hunk 1 above)
Comment (ArthurRab) [reply util/run.bzl]: You're right. Removed.

File: package_managers/download_pkgs.bzl (hunk 2 above)
[the remark this answers is not in the document]
Comment (ArthurRab) [reply package_managers/download_pkgs.bzl]: Done

File: util/run.bzl (hunk 1 above)
Comment (xingao267) [reply util/run.bzl]: I think there are more room for cleanup. We could also remove "image_name" from _run_and_extract, and container_run_and_extract does not need to call _rename_image anymore. Please test though.

File: package_managers/download_pkgs.bzl [hunk 7]
```diff
@@ -72,24 +72,38 @@ def _impl(ctx, image_tar=None, packages=None, additional_repos=None, output_exec
...
-    # docker_build rules always generate an image named 'bazel/$package:$name'.
-    builder_image_name = "bazel/%s:%s" % (image_tar.owner.package,
-                                          image_tar.basename.split(".tar")[0])
-
     # Generate a shell script to run apt_get inside this docker image.
     # TODO(tejaldesai): Replace this by docker_run rule
     build_contents = """\
 #!/bin/bash
 set -ex
+
```
Comment (xingao267) [inline package_managers/download_pkgs.bzl]: Did you look at if it's possible to reuse this part of the script across the three files?

**Comment by ArthurRab** [comment]
Moved to different branch for this fix.

**Comment by xingao267** [comment]
The review continues at https://github.com/ArthurRab/base-images-docker/pull/3/files

**Pull request closed without merging**